main.rb

```diff
@@ -0,0 +1,198 @@
+# frozen_string_literal: true
+
+require 'net/http'
+require 'json'
+require 'date'
+require 'colored'
+
+def env_has_key(key)
+  !ENV[key].nil? && ENV[key] != '' ? ENV[key] : abort("Missing #{key}.".red)
+end
+
+def get_env_variable(key)
+  return (ENV[key] == nil || ENV[key] == "") ? nil : ENV[key]
+end
+
+MINUTES_IN_A_DAY = 1440
+$username = env_has_key('AC_TESTINIUM_USERNAME')
+$password = env_has_key('AC_TESTINIUM_PASSWORD')
+$plan_id = env_has_key('AC_TESTINIUM_PLAN_ID')
+$ac_max_failure_percentage = (get_env_variable('AC_TESTINIUM_MAX_FAIL_PERCENTAGE') || 0).to_i
+$company_id = get_env_variable('AC_TESTINIUM_COMPANY_ID')
+$env_file_path = env_has_key('AC_ENV_FILE_PATH')
+$each_api_max_retry_count = env_has_key('AC_TESTINIUM_MAX_API_RETRY_COUNT').to_i
+timeout = env_has_key('AC_TESTINIUM_TIMEOUT').to_i
+date_now = DateTime.now
+$end_time = date_now + Rational(timeout, MINUTES_IN_A_DAY)
+$time_period = 30
+
+def get_parsed_response(response)
+  JSON.parse(response, symbolize_names: true)
+rescue JSON::ParserError, TypeError => e
+  puts "\nJSON expected but received: #{response}".red
+  puts "Error Message: #{e}".red
+  exit(1)
+end
+
+def calc_percent(numerator, denominator)
+  if !(denominator >= 0)
+    puts "Invalid numerator or denominator numbers".red
+    exit(1)
+  elsif denominator == 0
+    return 0
+  else
+    return numerator.to_f / denominator.to_f * 100.0
+  end
+end
+
+def check_timeout()
+  puts "Checking timeout...".yellow
+  now = DateTime.now
+
+  if now > $end_time
+    puts "Timeout exceeded! If you want to allow more time, please increase the AC_TESTINIUM_TIMEOUT input value.".red
```

Comment:
"Timeout exceeded" expression is not completely correct here. We can make more clear like;
```suggestion
    puts "Execution timed out! If you want to allow more time, please increase the AC_TESTINIUM_TIMEOUT input value.".red
```

Reply:
Actually this error message is a used error message. For example:
- https://help.mulesoft.com/s/question/0D52T00004mXXMoSAO/troubleshooting-timeout-exceeded-for-https-post

I am not sure the given suggestion is suitable here because I used this term because the "timeout" value given by the user as input was exceeded.